modules/openstack/limits.go

```diff
@@ -151,3 +161,73 @@ func (o *OpenStack) CreateRegisteredLimit(
 
 	return limitID, nil
 }
+
+// DeleteRegisteredLimit - delete limit from keystone
+func (o *OpenStack) DeleteRegisteredLimit(
+	log logr.Logger,
+	registeredLimitID string,
+) error {
+    log.Info(fmt.Sprintf("Deleting registered limit %s", registeredLimitID))
+	err := registeredlimits.Delete(o.osclient, registeredLimitID).ExtractErr()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//GetRegisteredLimit - Get existing registered limit by ID
+func (o *OpenStack) GetRegisteredLimit(
+	log logr.Logger,
+	registeredLimitID string,
+) (RegisteredLimit, error) {
+    log.Info(fmt.Sprintf("Fetching registered limit %s", registeredLimitID))
+	registered_limit, err := registeredlimits.Get(o.osclient, registeredLimitID).Extract()
+    if err != nil {
+        return _, err
+    }
+    return registered_limit, nil
+}
+
+//ListRegisteredLimitsByResourceName - List all registered limits filtered by resource name
+func (o *OpenStack) ListRegisteredLimitsByResourceName(
+	log logr.Logger,
+	resourceName string,
+) ([]RegisteredLimit, error) {
+    listOpts := registeredlimits.ListOpts{
+		ResourceName: resourceName,
+	}
+
+	log.Info(fmt.Sprintf("Fetching registered limit %s", resourceName))
+	allPages, err := registeredlimits.List(o.osclient, listOpts).AllPages()
+	if err != nil {
+		return _, err
+	}
+
+	allLimits, err := limits.ExtractRegisteredLimits(allPages)
```

Comment:
that should be `registeredlimits.ExtractRegisteredLimits(allPages)`

Reply:
Done